src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +91,50 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  /**
+   * Updates the workflow version and alerts the Dockstore User with success
+   * or failure.
+   *
+   * @private
+   * @memberof ViewWorkflowComponent
+   */
+  private updateWorkflowToSnapshot(version): void {
```

Comment:
All of this should be moved to a service.  The only thing in this file should be:
```
snapshotVersion() {
   this.potatoService.snapshotVersion(parameters);
}
```
and something to get the enum

Also, add type to the parameter

Reply:
I think this is the kind of thing that would be useful in a tutorial/talk/whatever type of session we have that was brought up by David in retro. Knowing what a service is, when to use it, how it fits in with Angular and Dockstore.

Reply:
Thanks both! Going to do as part of https://github.com/dockstore/dockstore/issues/2646